### PR TITLE
add missing security rules about NULL check to doc

### PR DESCRIPTION
**Pull request by icy17**
##### Checklist

- [x] documentation is added or updated

### CODE REVIEW

File: doc/man3/SSL_CTX_set_verify.pod [hunk 1]
```diff
@@ -32,7 +32,7 @@ SSL_CTX_set_post_handshake_auth
 
 SSL_CTX_set_verify() sets the verification flags for B<ctx> to be B<mode> and
 specifies the B<verify_callback> function to be used. If no callback function
-shall be specified, the NULL pointer can be used for B<verify_callback>.
+shall be specified, the NULL pointer can be used for B<verify_callback>. B<ctx> B<must not> be NULL.
```
Comment (InfoHunter) [inline doc/man3/SSL_CTX_set_verify.pod:35]: This is not consistent with other uppercase "MUST NOT" phrases.

File: doc/man3/EVP_EncryptInit.pod [hunk 2]
```diff
@@ -352,16 +352,16 @@ can be set in the context's current state.
...
+I<type> is typically supplied by calling EVP_CIPHER_fetch(). I<type> may also be set
 using legacy functions such as EVP_aes_256_cbc(), but this is not recommended
 for new applications. I<key> is the symmetric key to use and I<iv> is the IV to
 use (if necessary), the actual number of bytes used for the key and IV depends
 on the cipher. The parameters I<params> will be set on the context after
 initialisation. It is possible to set all parameters to NULL except I<type> in
 an initial call and supply the remaining parameters in subsequent calls, all of
 which have I<type> set to NULL. This is done when the default cipher parameters
-are not appropriate.
+are not appropriate.
```
Comment (InfoHunter) [inline doc/man3/EVP_EncryptInit.pod:364]: No trailing space.

File: doc/man3/ASN1_STRING_length.pod [hunk 3]
```diff
@@ -28,11 +28,11 @@ ASN1_STRING_to_UTF8 - ASN1_STRING utility functions
 
 These functions allow an B<ASN1_STRING> structure to be manipulated.
 
-ASN1_STRING_length() returns the length of the content of I<x>.
+ASN1_STRING_length() returns the length of the content of I<x>. I<x> B<MUST NOT> be NULL.
```
Comment (InfoHunter) [inline doc/man3/ASN1_STRING_length.pod:31]: I checked some other places among the pod files, NULL should also be bold as "B\<NULL>"

**Comment by icy17** [comment]
Thanks for the feedback. I'll fix the issues.

File: doc/man3/ASN1_STRING_length.pod (hunk 3 above)
Comment (t8m) [reply doc/man3/ASN1_STRING_length.pod]: No! It should not.

File: doc/man3/ASN1_STRING_length.pod [hunk 4]
```diff
@@ -28,11 +28,11 @@ ASN1_STRING_to_UTF8 - ASN1_STRING utility functions
 
 These functions allow an B<ASN1_STRING> structure to be manipulated.
 
-ASN1_STRING_length() returns the length of the content of I<x>.
+ASN1_STRING_length() returns the length of the content of I<x>. I<x> B<MUST NOT> be B<NULL>.
```
Comment (t8m) [inline doc/man3/ASN1_STRING_length.pod:31]: Please revert the `NULL`->`B<NULL>` changes. @InfoHunter was wrong, NULL should not be put in bold.

**Review by t8m — CHANGES_REQUESTED** [review]

**Review by t8m — APPROVED** [review]

**Comment by FdaSilvaYY** [comment]
Typo : `rulse` in Commit Message

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
ping @openssl/committers for the second review

**Comment by beldmit** [comment]
BTW, why do we have ABI label for doc-only changes?..

**Review by nhorman — APPROVED** [review]

**Review by shahsb — APPROVED** [review]
LGTM.

**Comment by t8m** [comment]
Merged to the master branch. Thank you for your contribution.

**Pull request closed without merging**